README.md

```diff
@@ -18,52 +18,106 @@ The NTT manager contracts additionally keep track of the token decimals of the o
-1. **Sending**: A client calls on [`transfer`] to initiate an NTT transfer. The client must specify at minimum, the amount of the transfer, the recipient chain, and the recipient address on the recipient chain. [`transfer`] also supports a flag to specify whether the NTTManager should queue transfers if they are rate-limited. Clients can also include additional instructions to forward along to its peer NTT Manager on the recipient chain to execute. Depending on the mode, transfers are either "locked" or "burned". Once the transfer has been forwarded to the Transceiver, the NTTManager emits the following event:
+Ntt supports rate limiting on the sending. If a trnasfer is rate limited on the source chain and queueing is enabled, transfers are placed into a outbound queue that can be released after a 24-hr window via the [`completeInboundTransferQueue`] method.
+
+Transfers that are rate-limited on the destination chain are added to an InboundTransferQueue that can be released via the [`completeInboundTransferQueue`] method.
+
+If users bridge frequently between an arbitrary source chain and a given destination chain, the capacity could be exhausted quickly, leaving other users rate-limited, potentially delaying transfers. To mitigate this issue, we refilling the inbound rate limit on the source chain by the  equivalent amount of the outbound transfer. We refer to this colloquially as "Cancel Flow".
+
+### NTT Message Lifecycle (EVM)
+
+1. **Send**: A client calls on [`transfer`] to initiate an NTT transfer. The client must specify at minimum, the amount of the transfer, the recipient chain, and the recipient address on the recipient chain. [`transfer`] also supports a flag to specify whether the NTTManager should queue transfers if they are rate-limited. Clients can also include additional instructions to forward along to its peer NTT Manager on the recipient chain to execute. Depending on the mode, transfers are either "locked" or "burned". Once the transfer has been forwarded to the Transceiver, the NttManager emits the following event:
 
 ``` solidity
-emit TransferSent(recipient, _nttDenormalize(amount), recipientChain, seq);
+    event TransferSent(
+        bytes32 recipient, uint256 amount, uint256 fee, uint16 recipientChain, uint64 msgSequence
+    );
 ```
-2. **Rate Limiting**: NTT supports rate-limiting of tranfers based on a 24-hr sliding window. This is intended to be a defense-in-depth security measure to mitigate and localize risk. This is a security feature intended to protect large unintended transfers. If a transfer sent from the source chain is rate-limited, it is added to a queue of transfers. The following event is emitted:
+
+2. **Rate Limit**: NTT supports rate-limiting of tranfers based on a 24-hr sliding window. This is intended to be a defense-in-depth security measure to mitigate and localize risk.
+If a transfer sent from the source chain is rate-limited and queueing is enabled, it is added to a queue of transfers.
```

Comment:
NTT supports rate-limiting of transfers based on a configurable `_rateLimitDuration` sliding window. Rate limiting is a defense-in-depth security measure to mitigate and localize risk. By default, the deployment scripts set `_rateLimitDuration` to 24 hours. NTT supports configuring both outbound and inbound rate limits per chain deployment. The outbound limit is a single sending limit, while the inbound limits are per-chain. 

If a transfer initiated on the source chain is rate-limited, it is only added to a queue of transfers if the client sets the `shouldQueue` flag to `true`.  If a transfer is rate limited on the destination chain, it is always added to a queue.

Reply:
Consider also adding this additional detail:

"Inbound transfers are always queued since users pay relayers up front on the source chain to deliver messages once. Always queuing on the destination chain simplifies UX since relayers can always deliver messages successfully, and users don't have to manually retry delivery multiple times in case they hit rate limits."